integration-test/src/test/java/org/ehcache/integration/transactions/xa/XACacheTest.java

```diff
@@ -66,40 +67,61 @@
  */
 public class XACacheTest {
 
+  @Rule
+  public TemporaryFolder folder = new TemporaryFolder();
+
+  private File storagePath;
+
+  private CacheManager cacheManager = null;
+
+  private final TestTimeSource testTimeSource = new TestTimeSource();
+
+  private BitronixTransactionManager transactionManager;
+
   @Before
   public void setUpBtmConfig() throws Exception {
+    storagePath = folder.newFolder();
+
+    initTransactionManagerServices();
+
+    transactionManager = TransactionManagerServices.getTransactionManager();
+  }
+
+  private void initTransactionManagerServices() {
     TransactionManagerServices.getConfiguration()
-        .setLogPart1Filename(getStoragePath() + "/btm1.tlog")
-        .setLogPart2Filename(getStoragePath() + "/btm2.tlog")
+        .setLogPart1Filename(new File(getStoragePath(), "btm1.tlog").getAbsolutePath())
+        .setLogPart2Filename(new File(getStoragePath(), "btm2.tlog").getAbsolutePath())
         .setServerId("XACacheTest")
         .setGracefulShutdownInterval(0);
   }
 
   @After
   public void tearDownBtm() throws Exception {
+    if(cacheManager != null) {
+      cacheManager.close();
+    }
+    transactionManager.shutdown();
```

Comment:
Adding this makes the following block useless? Although it seems the following block is _safer_ in a way.
@lorban Care to comment?

Reply:
Right, the `transactionManager.shutdown()` call is redundant with the `if ()` block below.

Reply:
Cool thanks